packages/cdk-cloudfront-authorization/src/authorizations.ts

```diff
@@ -39,20 +39,22 @@ export interface AuthorizationProps {
   readonly logLevel?: LogLevel;
   readonly oauthScopes?: OAuthScope[];
   readonly cookieSettings?: Record<string, string>;
+  readonly identityProviders?: UserPoolClientIdentityProvider[];
 }
 
 export abstract class Authorization extends Construct {
   public readonly redirectPaths: RedirectPaths;
   public readonly signOutUrlPath: string;
   public readonly authFlow: AuthFlow;
+  public readonly userPoolClient: IUserPoolClient;
```

Comment:
```typescript
authorization.userPoolClient.node.addDependency(azureProvider);
```
Do we really have to do that?

I can't find any information about this in the [documentation](https://github.com/aws/aws-cdk/tree/37fdc94a9fdcaf37e02e66d6f1bf3c5501aee1a0/packages/%40aws-cdk/aws-cognito#identity-providers)

Reply:
I just tried it without specifying the dependency and received this error message:
```
21:42:12 | CREATE_FAILED        | AWS::Cognito::UserPoolClient                    | UserPool2/UserPoolClient
The provider Azure does not exist for User Pool eu-central-1_XXXXXXX. (Service: AWSCognitoIdentityProviderService; Status Code: 400; Error Code: InvalidPar
ameterException; Request ID: xxxxxxxxxxx; Proxy: null)
```

There is an example, where they also added the dependency manually: https://github.com/aws-samples/cdk-cognito-idp/blob/master/lib/cognito-idp-stack.ts
```
// Make sure the user pool client is created after the IDP
userPoolClient.node.addDependency(idp);
```

From my point of view, it makes sence that it is necessary. First, the Identity Provider is specified.
```
const idp = new cognito.UserPoolIdentityProviderFacebook(this, 'FacebookIDP', {
   // ...
}
```
Next, a constant is passed to the UserPoolClient. There is no reference to the IdentityProvider created before. Therefore, CDK can't build the dependency on its own.
`supportedIdentityProviders: [cognito.UserPoolClientIdentityProvider.FACEBOOK]`

Reply:
Ah okay. I didn't know that sample. Then it's okay, but we should point it out in the README

Reply:
I've enhanced the README file and described "addDependency".